start.R

```diff
@@ -297,6 +297,9 @@ modeltestRunsUsed <- 0
 
 testOneRegi_region <- ""
 
+# Save whether model is locked before runs are started
+model_was_locked <- file.exists(".lock")
+
```

Comment:
This breaks with the new locking.

Reply:
@mikapfl: I introduced this already [in March](https://github.com/remindmodel/remind/commit/a7c1600c41c78c4d81efc502cf29640cf3c440ad). Once the new locking is in place, we can simply delete it. The problem is that killing a job does not remove it from the queue, so the `.lock` file remains there even if no run is present that will unlock it at some point. The new lock mechanism doesn't have this problem, so we can simply delete it once it is implemented.

Reply:
I see, it was introduced yesterday, and only moved in this commit. Anyway, it will break. model_lock and model_unlock don't guarantee that the lock file will be present when the model is locked, or that the lock file will be removed when the model is unlocked.

Reply:
No, it was introduced in March before we even started discussing the lock problem, then moved yesterday, and moved again today. I think "break" is a bit too hard :) It may give you inaccurate information if the `.lock` file is there although the model is using a different locking mechanism and it may not be locked in reality. Yes, let us not forget it, but currently I decided to merge to fix the actual bug that breaks all coupled runs…

Reply:
well, all functionalities provided by `model_was_locked` (the information given to the user) will be inaccurate, so it breaks this thing, but it doesn't break the model. :wink: 

Sure, sorry, my `git blame` is weak and I started this discussion in the wrong PR/issue. As you said, it is not catastrophic to have wrong information for the user and your plan sounds good. But it is a bit unfortunate that things were implemented which circumvent the API. This model_lock/model_unlock API is there as an abstraction to separate concerns, and proper engineering would have been to introduce a `gms::model_locked` function (or something along those lines) and use that in the model so that all things related to locks are in the same library (and can e.g. be tested there).